homeassistant/components/binary_sensor/pilight.py

```diff
@@ -0,0 +1,190 @@
+"""
+Support for Pilight binary sensors.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/binary_sensor.pilight/
+"""
+import datetime
+import logging
+
+import voluptuous as vol
+from homeassistant.components import pilight
+from homeassistant.components.binary_sensor import (
+    PLATFORM_SCHEMA,
+    BinarySensorDevice,
+)
+from homeassistant.const import (
+    CONF_DISARM_AFTER_TRIGGER,
+    CONF_NAME,
+    CONF_PAYLOAD,
+    CONF_PAYLOAD_OFF,
+    CONF_PAYLOAD_ON,
+    STATE_OFF,
+    STATE_ON,
+)
+from homeassistant.helpers import config_validation as cv
+from homeassistant.helpers.event import track_point_in_time
+from homeassistant.util import dt as dt_util
+
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_VARIABLE = 'variable'
+
+DEFAULT_NAME = 'Pilight Binary Sensor'
+DEPENDENCIES = ['pilight']
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_VARIABLE): cv.string,
+    vol.Required(CONF_PAYLOAD): vol.Schema(dict),
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    vol.Optional(CONF_PAYLOAD_ON, default='on'): cv.string,
```

Comment:
The 'CONF_DISARM_AFTER_TRIGGER' is really a cv.boolean, but 'CONF_PAYLOAD_ON' can be any string (e.g. "opened", "high", etc.) so I think it's work only with string, because at state generation the 'self._state = (value == self._on_value)' make a string comparison.